src/components/pip/pictue-in-picture.tsx

```diff
@@ -0,0 +1,155 @@
+ * @extends {Component}
+ */
+@redux.connect(mapStateToProps, utils.bindActions({...reducers.shell.actions, ...reducers.engine.actions}))
+@withText({
+  pictureInPictureText: 'controls.pictureInPicture',
+  pictureInPictureExitText: 'controls.pictureInPictureExit'
+})
+class PictureInPicture extends Component<PictureInPictureDualScreenProps>  {
+  buttonContainerRef:HTMLButtonElement | null = null
+
+  /**
+   * Creates an instance of PictureInPicture.
+   * @memberof PictureInPicture
+   */
+  constructor(props: any) {
+    super(props);
+  }
+  componentDidMount() {
+    this.buttonContainerRef!.parentNode!.nextSibling!.textContent = this.props.pictureInPictureText|| ""
+  }
+
+  componentDidUpdate() {
+    if(this.props.playerType === PlayerType.VIDEO) {
+      const videoPlayer:KalturaPlayerTypes.Player = this.props.player as KalturaPlayerTypes.Player;
+      videoPlayer.ui.store.dispatch(reducers.engine.actions.updateIsInPictureInPicture(false));
+    }
+  }
+
+  /**
+   * toggle pip
+   * @returns {void}
+   *
+   * @memberof PictureInPicture
+   */
+  togglePip = (): void => {
+    const player = this.props.player;
+    if (!player) {
+      return;
+    }
+
+    if(this.props.playerType === PlayerType.IMAGE) {
+      return
+    }
+
+    if (this.isSomePlayerInPip()) {
+      this.exitPlayerInPip();
+
+      const currentClassName = this.buttonContainerRef?.className.replace(" " + style.isInPictureInPicture, "") as string;
+      this.updateParams(currentClassName, this.props.pictureInPictureText|| "")
+    } else {
+      //@ts-ignore
+      player.enterPictureInPicture();
+
+      const currentClassName = this.buttonContainerRef?.className.concat(" " + style.isInPictureInPicture || "") as string;
+      this.updateParams(currentClassName, this.props.pictureInPictureExitText|| "")
+    }
+  }
+
+  private updateParams = (className:string, ariaLabel:string) => {
+    this.buttonContainerRef?.setAttribute('class', className)
+    this.buttonContainerRef?.setAttribute('aria-label', ariaLabel)
+    this.buttonContainerRef!.parentElement!.nextElementSibling!.textContent = ariaLabel;
+  }
+
+  public exitPlayerInPip = () => {
+    return this.props.dualScreenPlayers.find(dualScreenPlayer => {
+      try{
+        //@ts-ignore
+        if(dualScreenPlayer.player.isInPictureInPicture()){
+          //@ts-ignore
+          dualScreenPlayer.player.exitPictureInPicture();
+          return
+        }
+      }
+      catch {
+      }
+    });
+  };
+
+  public isSomePlayerInPip = () => {
+    return this.props.dualScreenPlayers.find(dualScreenPlayer => {
+      try{
+        //@ts-ignore
+        if(dualScreenPlayer.player.isInPictureInPicture()){
+          return true
+        }
+      }
+      catch {
+      }
+    });
+  };
+
+  /**
+   * render component
+   *
+   * @returns {React$Element} - component element
+   * @memberof PictureInPicture
+   */
+  render() {
+    const isSomePlayerInPip = this.isSomePlayerInPip()
+    return (
```

Comment:
originar `PictureInPicture` component from `playkit-js-ui` uses `ButtonControl`
https://github.com/kaltura/playkit-js-ui/blob/87ca479e640ef7cd7960e22dbe4fb665584f8aed/src/components/picture-in-picture/picture-in-picture.tsx#L156
and `Icon`
https://github.com/kaltura/playkit-js-ui/blob/87ca479e640ef7cd7960e22dbe4fb665584f8aed/src/components/picture-in-picture/picture-in-picture.tsx#L165-L166
It seems like make sense use same approach for custom `PiP` component